pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -237,6 +237,10 @@ func (m *Multicluster) initializeCluster(cluster *multicluster.Cluster, kubeCont
 	if m.configController != nil && features.EnableAmbientControllers {
 		m.configController.RegisterEventHandler(gvk.AuthorizationPolicy, kubeRegistry.AuthorizationPolicyHandler)
 		m.configController.RegisterEventHandler(gvk.PeerAuthentication, kubeRegistry.PeerAuthenticationHandler)
+		if configCluster && m.serviceEntryController != nil {
+			// configController is used by ambient service entry handler to get workload entries
+			m.serviceEntryController.AppendServiceHandler(kubeRegistry.ServiceEntryHandler)
```

Comment:
It's starting to feel weird for the ambientindex to be a subcomponent of the kube controller...

Reply:
no actual suggestion here.